### PR TITLE
test: Test cases for hidePresentationOnJoin parameter

### PR DESCRIPTION
**Pull request by gabriellpr**
### What does this PR do?
Add test cases for the hidePresentationOnJoin parameter: 

 
  *  Start and stop screen sharing
  *  Show external videos
  *  Pin and unpin shared notes
  *  Upload large presentation
  *  Change layout
  *  Return from breakout rooms

### CODE REVIEW

File: bigbluebutton-tests/playwright/parameters/parameters.spec.js [hunk 1]
```diff
@@ -479,14 +479,58 @@ test.describe.parallel('Custom Parameters', { tag: '@ci' }, () => {
     });
   });
 
-  test.describe.parallel('Presentation', () => {
+  test.describe.parallel('Hide Presentation On Join Edge Cases', () => {
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/parameters.spec.js]: keep this test suite name generic and the test specs with the edge case name only to avoid excessive naming on tests:
```
  [Chromium] › parameters/parameters.spec.js:483:5 › Custom Parameters › Hide Presentation On Join Edge Cases › Hide Presentation on join
  [Chromium] › parameters/parameters.spec.js:490:5 › Custom Parameters › Hide Presentation On Join Edge Cases › Hide Presentation on join Screenshare
  [Chromium] › parameters/parameters.spec.js:497:5 › Custom Parameters › Hide Presentation On Join Edge Cases › Hide presentation on join External video
  [Chromium] › parameters/parameters.spec.js:504:5 › Custom Parameters › Hide Presentation On Join Edge Cases › Hide presentation on join Pin and unpin shared notes
  [Chromium] › parameters/parameters.spec.js:511:5 › Custom Parameters › Hide Presentation On Join Edge Cases › Hide presentation on join Change Layout
  [Chromium] › parameters/parameters.spec.js:518:5 › Custom Parameters › Hide Presentation On Join Edge Cases › Hide presentation on join return from breakouts
  [Chromium] › parameters/parameters.spec.js:525:5 › Custom Parameters › Hide Presentation On Join Edge Cases › Hide presentation on join, upload large presentation
```

test spec name suggestions:
- "joining with the parameter" (default test)
- "after sharing screen"
- "after sharing external video"
- ...

```suggestion
  test.describe.parallel('Hide Presentation on join', () => {
```

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 2]
```diff
@@ -158,6 +159,144 @@ class CustomParameters extends MultiUsers {
...
+    await this.modPage.hasElement(e.actions, 'should display the actions button');
+    await this.modPage.hasElement(e.restorePresentation, 'should display the restore presentation button for the moderator');
+    await this.userPage.hasElement(e.restorePresentation, 'should display the restore presentation button for the attendee');
+    await this.userPage.wasRemoved(e.whiteboard, 'should not display the whiteboard for the attendee');
+
+    await this.modPage.waitAndClick(e.actions);
+    await this.modPage.waitAndClick(e.managePresentations);
+    await this.modPage.hasElement(e.presentationFileUpload, 'should display the presentation space for uploading a new file, when the manage presentations is opened');
+  
+    await this.modPage.page.setInputFiles(e.presentationFileUpload, path.join(__dirname, `../core/media/${e.pdfFileName}`));
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js]: why not use the `uploadSinglePresentation` util function here?

File: bigbluebutton-tests/playwright/parameters/parameters.spec.js [hunk 3]
```diff
@@ -479,14 +479,58 @@ test.describe.parallel('Custom Parameters', { tag: '@ci' }, () => {
...
     });
 
+    test('Hide Presentation on join Screenshare', async ({ browser, context, page }) => {
+      const customParam = new CustomParameters(browser, context);
+      await customParam.initModPage(page, true, { joinParameter: c.hidePresentationOnJoin });
+      await customParam.initUserPage(true, context, { useModMeetingId: true, joinParameter: c.hidePresentationOnJoin });
+      await customParam.hidePresentationOnJoinScreenshare();
+    });
+
+    test('Hide presentation on join External video', { tag: '@flaky' }, async({ browser, context, page }) => {
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/parameters.spec.js]: add a comment with the flaky reason + use `linkIusse(21589)` (just like https://github.com/antonbsa/bigbluebutton/blob/v3.0.x-release/bigbluebutton-tests/playwright/screenshare/screenshare.spec.js#L17-L18)

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 4]
```diff
@@ -158,6 +159,144 @@ class CustomParameters extends MultiUsers {
...
+    await this.modPage.hasElement(e.restorePresentation, 'should display the restore presentation button for the moderator');
+    await this.userPage.hasElement(e.restorePresentation, 'should display the restore presentation button for the attendee');
+    await this.userPage.wasRemoved(e.whiteboard, 'should not display the whiteboard for the attendee');
+
+    await this.modPage.waitAndClick(e.sharedNotes);
+    await this.modPage.waitAndClick(e.notesOptions);
+    await this.modPage.waitAndClick(e.pinNotes);
+    await this.modPage.hasElement(e.unpinNotes, 'should display the unpin notes button');
+    await this.userPage.hasElement(e.minimizePresentation, 'should display the minimize presentation button for the attendee');
+    await this.modPage.waitAndClick(e.unpinNotes);
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js:215]: before clicking to unpin, certify that the notification is closed. otherwise, it fails when trying to perform the click:

image

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 5]
```diff
@@ -158,6 +159,144 @@ class CustomParameters extends MultiUsers {
...
+  async hidePresentationOnJoinScreenshare() {
+    await this.modPage.hasElement(e.actions, 'should display the actions button');
+    await this.modPage.hasElement(e.restorePresentation, 'should display the restore presentation button for the moderator');
+    await this.userPage.hasElement(e.restorePresentation, 'should display the restore presentation button for the attendee');
+    await this.userPage.wasRemoved(e.whiteboard, 'should not display the whiteboard for the attendee');
+
+    await this.modPage.waitAndClick(e.startScreenSharing);
+    await this.modPage.hasElement(e.screenShareVideo, '', 20000);
+    await this.modPage.hasElement(e.stopScreenSharing);
+    await this.userPage.hasElement(e.screenShareVideo, 'should display the screenshare element');
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js]: why not use the `startScreenshare` screenshare util function?

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 6]
```diff
@@ -158,6 +159,144 @@ class CustomParameters extends MultiUsers {
...
+    await this.userPage.wasRemoved(e.whiteboard, 'should not display the whiteboard for the attendee');
+
+    await this.modPage.waitAndClick(e.optionsButton);
+    await this.modPage.waitAndClick(e.manageLayoutBtn);
+    await this.modPage.waitAndClick(e.focusOnVideo);
+    await this.modPage.waitAndClick(e.updateLayoutBtn);
+    await this.modPage.closeAllToastNotifications();
+    await this.modPage.wasRemoved(e.toastContainer);
+
+    //await checkScreenshots(this, 'should be the grid layout', [e.webcamContainer, e.webcamMirroredVideoContainer], 'grid-layout');
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js]:
```suggestion
```

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 7]
```diff
@@ -158,6 +159,144 @@ class CustomParameters extends MultiUsers {
...
+    
+    await this.modPage.hasElement(e.presentationUploadProgressToast, 'should display the toast presentation upload progress after confirming the presentation to be uploaded');
+    await sleep(10000);
+
+    const wbBox = await this.modPage.getLocator(e.whiteboard);
+    await expect(wbBox).toHaveScreenshot('moderator-with-minimalist-large-presentation.png');
+
+    await this.modPage.hasElement(e.minimizePresentation, 'should display the restore presentation button for the moderator');
+    await this.userPage.hasElement(e.minimizePresentation, 'should display the restore presentation button for the attendee');
+    await this.userPage.hasElement(e.whiteboard, 'should not display the whiteboard for the attendee');
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js]:
```suggestion
    await this.userPage.hasElement(e.whiteboard, 'should display the whiteboard for the attendee');
```

**Review by antonbsa — CHANGES_REQUESTED** [review]
added a few change requests. tests passing (besides  the pin and unpin case I've mentioned below)

**Review by antonbsa — APPROVED** [review]
